### PR TITLE
Change autolink placeholder to something less likely

### LINKED ISSUE #1091: ex_doc 0.21.2: "@spec x_p0_x() :: any" crashes mix docs
Erlang/OTP 22 [erts-10.4.2] [source] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1]
Elixir 1.9.1 (compiled with Erlang/OTP 22)
ex_doc 0.21.2

This compiles fine but `mix docs` crashes:

`defmodule A do
  @spec x_p0_x() :: any
  def x_p0_x() do
    42
  end
end`

Seems to be because of the *_p0_* in the function name:

>  ** (KeyError) key :_p0_ not found in: %{_p1_: "<a href=\"https://hexdocs.pm/elixir/typespecs.html#basic-types\">any</a>"}
        (stdlib) :maps.get(:_p0_, %{_p1_: "<a href=\"https://hexdocs.pm/elixir/typespecs.html#basic-types\">any</a>"})
        (ex_doc) lib/ex_doc/formatter/html/autolink.ex:334: anonymous fn/2 in ExDoc.Formatter.HTML.Autolink.replace_placeholders/2
        (elixir) lib/regex.ex:731: Regex.apply_list/5
        (elixir) lib/regex.ex:726: Regex.apply_list/5
        (elixir) lib/regex.ex:668: Regex.do_replace/4
        (elixir) lib/enum.ex:1336: Enum."-map/2-lists^map/1-0-"/2
        (ex_doc) lib/ex_doc/formatter/html/autolink.ex:205: anonymous fn/6 in ExDoc.Formatter.HTML.Autolink.all_typespecs/2
        (elixir) lib/enum.ex:1948: Enum."-reduce/3-lists^foldl/2-0-"/3

### PR DESCRIPTION
**Pull request by wojtekmach**
Closes #1091

### CODE REVIEW

File: lib/ex_doc/formatter/html/autolink.ex [hunk 1]
```diff
@@ -321,17 +321,23 @@ defmodule ExDoc.Formatter.HTML.Autolink do
     end
   end
 
+  @placeholder "ó"
```
Comment (josevalim) [inline lib/ex_doc/formatter/html/autolink.ex]: Let’s use $. But then we will need to escape it in the regex.

Comment (wojtekmach) [reply lib/ex_doc/formatter/html/autolink.ex]: Good call. I got some failing tests on CI, which I didn’t get locally, so there might still be something wrong.

Comment (josevalim) [reply lib/ex_doc/formatter/html/autolink.ex]: Never mind. It has to be a valid identifier which is why ó fails on v1.7. We will have to make it something longer then. A mixture of lowercase, uppercase, underscore and digits.

**Comment by wojtekmach** [comment]
@josevalim updated!

**Pull request merged**